apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/checkApolloDuplicateTargetNames.kt

```diff
@@ -0,0 +1,58 @@
+@file:JvmName("-checkApolloDuplicateTargetNames")
+
+package com.apollographql.apollo3.compiler
+
+import com.apollographql.apollo3.annotations.ApolloInternal
+import com.apollographql.apollo3.ast.GQLDirective
+import com.apollographql.apollo3.ast.GQLEnumTypeDefinition
+import com.apollographql.apollo3.ast.GQLInputObjectTypeDefinition
+import com.apollographql.apollo3.ast.GQLInterfaceTypeDefinition
+import com.apollographql.apollo3.ast.GQLObjectTypeDefinition
+import com.apollographql.apollo3.ast.GQLScalarTypeDefinition
+import com.apollographql.apollo3.ast.GQLTypeDefinition
+import com.apollographql.apollo3.ast.GQLUnionTypeDefinition
+import com.apollographql.apollo3.ast.Issue
+import com.apollographql.apollo3.ast.Schema
+import com.apollographql.apollo3.ast.findTargetName
+
+@ApolloInternal
+fun checkApolloDuplicateTargetNames(schema: Schema): List<Issue> {
+  val issues = mutableListOf<Issue>()
+
+  val visitedTypes = mutableListOf<GQLTypeDefinition>()
+  for (type in schema
+      .typeDefinitions
+      .values
+  ) {
+    val targetName = type.directives.findTargetName(schema)
+    if (targetName == null) {
+      visitedTypes += type
```

Comment:
Somewhat a personal preference but I'm not a fan of the augmentedAssigment operator on mutable lists as it actually doesn't assign anything 🤷